promql/engine.go

```diff
@@ -1628,7 +1628,9 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, annotations.Annotatio
 					maxt := ts - offset
 					mint := maxt - selRange
 					floats, histograms = ev.matrixIterSlice(it, mint, maxt, floats, histograms)
+					floats = ApplyBinaryOps(sel.BinaryOps, floats)
```

Comment:
Wouldnt it be clearer if this was part of "matrixIterSlice" itself already, right now we call it twice outside of that function but pretty much immediately after. To me it would be clearer abstraction if we would filter inside that already maybe?